storage/innobase/row/row0merge.cc

```diff
@@ -3551,7 +3551,7 @@ row_merge_sort(
 	*/
 #ifndef __sun__
 	/* Progress report only for "normal" indexes. */
-	if (dup && !(dup->index->type & DICT_FTS)) {
+	if (update_progress) {
```

Comment:
This is partly repurposing the parameter `update_progress`. That may be fine, but shouldn’t we retain the condition on `dup->index->type` or move it to those callers that currently passes `update_progress=true`? Or add debug assertions at the start of this function:
```c++
ut_ad(!update_progress || dup);
ut_ad(!update_progress || !dup || !(dup->index->type & DICT_FTS));
```